source/class/qx/tool/compiler/app/ManifestFont.js

```diff
@@ -0,0 +1,171 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2023 Zenesis Limited https://www.zenesis.com
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * John Spackman (@johnspackman)
+
+************************************************************************ */
+
+const fs = require("fs");
+const path = require("path");
+
+/**
+ * Represents a font as defined in the Manifest.json's
+ */
+qx.Class.define("qx.tool.compiler.app.ManifestFont", {
+  extend: qx.core.Object,
+
+  construct(name) {
+    super();
+    this.setName(name);
+  },
+
+  properties: {
+    /** The name of the font - this is the key in Manifest.json provides.fonts */
+    name: {
+      check: "String"
+    },
+
+    defaultSize: {
+      init: null,
+      nullable: true,
+      check: "Integer"
+    },
+
+    comparisonString: {
+      init: null,
+      nullable: true,
+      check: "String"
+    },
+
+    urls: {
+      init: null,
+      check: "Array",
+      nullable: true
+    },
+
+    sources: {
+      init: null,
+      check: "Array",
+      nullable: true
+    },
+
+    glyphs: {
+      init: null,
+      nullable: true,
+      check: "String"
+    },
+
+    family: {
+      init: null,
+      nullable: true,
+      check: "Array"
+    }
+  },
+
+  members: {
+    /** @type{Map<String,Object>} font data required by the Qooxdoo app at runtime */
+    __fontData: null,
+
+    /**
+     * Updates this from the data in the Manifest.json
+     *
+     * @param {Object} data the data from Manifest
+     * @param {qx.tool.compiler.app.Library} library the library
+     */
+    async updateFromManifest(data, library) {
+      let toSet = {};
+      [
+        "defaultSize",
+        "comparisonString",
+        "urls",
+        "sources",
+        "glyphs",
+        "family"
+      ].forEach(name => {
+        if (data.hasOwnProperty(name)) {
+          toSet[name] = data[name];
+        }
+      });
```

Comment:
```suggestion
      ].filter(name => data.hasOwnProperty(name))
      .forEach(name => toSet[name] = data[name]);
```

Reply:
this is introducing a bug, and the reason why I wrote it as it was and not with `Object.assign`